File: Aminolat101/ViewController.m
//
//  ViewController.m
//  CollectionViewWithSearchBar
//
//  Created by Bean Bob on 5/29/15.
//  Copyright (c) 2015 Bean Bob. All rights reserved.
//

#import "ViewController.h"
@import Firebase;
@import FirebaseAuth;
@import GoogleSignIn;



@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
    [GIDSignIn sharedInstance].clientID = [FIRApp defaultApp].options.clientID;
    [GIDSignIn sharedInstance].delegate = self;
    [GIDSignIn sharedInstance].uiDelegate = self;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


-(void)signIn:(GIDSignIn *) signIn didSignInForUser:(GIDGoogleUser *)user withError:(NSError *)error {
    
    if (user) {
        NSString * welcomeMessage = [NSString stringWithFormat: @"Welcome To FireBase, %@"];
        UIAlertView *alert = [[UIAlertView alloc]
                              initWithTitle:@"Firebase" message:welcomeMessage delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil];
    } else {
        
        NSLog(@"%@", error.localizedDescription);
    }
    
    }
    
-(IBAction)SignOutClicked:(id)sender {
    
    FIRAuth *firebaseAuth = [FIRAuth auth];
    NSError * signOutError;
    BOOL status = [firebaseAuth signOut:&signOutError];
    if (!status) {
        NSLog(@"Error Signing Out: %@", signOutError);
        return;
    }
    
//    _signInButton.enabled = true;
//    _signOutButton.enabled = false;

}


@end

File: Aminolat101/ViewController.swift
//
//  ViewController.swift
//  Pinterest
//
//  Created by Mikkel Jensen on 08/06/16.
//  Copyright © 2016 Razeware LLC. All rights reserved.
//


import UIKit
import Firebase

class ViewController: UIViewController, UISearchControllerDelegate, UISearchResultsUpdating, UISearchBarDelegate {
  
  
  
  var searchController : UISearchController!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    self.searchController = UISearchController(searchResultsController:  nil)
    
    self.searchController.searchResultsUpdater = self
    self.searchController.delegate = self
    self.searchController.searchBar.delegate = self
    
    self.searchController.hidesNavigationBarDuringPresentation = false
    self.searchController.dimsBackgroundDuringPresentation = true
    
    self.navigationItem.titleView = searchController.searchBar
    
    self.definesPresentationContext = true
  }
  
  func updateSearchResultsForSearchController(searchController: UISearchController) {
    
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
  
  func setStatusBarBackgroundColor(color: UIColor) {
    
    guard  let statusBar = UIApplication.sharedApplication().valueForKey("statusBarWindow")?.valueForKey("statusBar") as? UIView else {
      return
      
    }
    
    statusBar.backgroundColor = UIColor.blueColor()
  }
}


/*
  //Changing Status Bar
   func preferredStatusBarStyle() -> UIStatusBarStyle {
    
    //LightContent
    return UIStatusBarStyle.LightContent
    
    //Default
    //return UIStatusBarStyle.Default
    
  }
*/  
